File: src/assignment3/src/goal_manager_node.cpp
#include "ros/ros.h"
#include "assignment3/UpdateGoals.h"

bool update_goals(assignment3::UpdateGoals::Request &req, assignment3::UpdateGoals::Response &res) {
    // Update and prioritize goals based on feedback
    assignment3::Goal goal;
    goal.description = "New goal";
    goal.priority = 1;
    res.goals.primary_goals.push_back(goal);
    return true;
}

int main(int argc, char **argv) {
    ros::init(argc, argv, "goal_manager_node");
    ros::NodeHandle n;

    ros::ServiceServer service = n.advertiseService("update_goals", update_goals);
    ros::spin();

    return 0;
}

File: src/assignment3/src/goal_manager_node.py
import rospy
from assignment3.srv import UpdateGoals, UpdateGoalsResponse
from assignment3.msg import Goal

def update_goals(req):
    # Update and prioritize goals based on feedback
    goal = Goal()
    goal.description = "New goal"
    goal.priority = 1
    res = UpdateGoalsResponse()
    res.goals.primary_goals.append(goal)
    return res

def goal_manager_server():
    rospy.init_node('goal_manager_node')
    service = rospy.Service('update_goals', UpdateGoals, update_goals)
    rospy.spin()

if __name__ == "__main__":
    goal_manager_server()
